lxd/storage_pools_config.go

```diff
@@ -72,6 +73,13 @@ var storagePoolConfigKeys = map[string]func(value string) error{
 	// valid drivers: ceph, lvm
 	"volume.size": shared.IsSize,
 
+	// valid drivers: all
+	"volume.snapshots.expiry": func(value string) error {
+		// Validate expression
+		_, err := shared.GetSnapshotExpiry(time.Time{}, value)
+		return err
+	},
```

Comment:
I don't think we want a `volume.snapshots.expiry` key at all. Only a `snapshots.expiry` key on volumes themselves.

Reply:
Fixed

Reply:
@stgraber does the `snapshots.expiry` property for instances get applied from an expanded profile? Would the equivalent for custom volumes be to allow the `volume.snapshots.expiry` key to be set for all new volumes on that pool?

Reply:
@tomponline it would, though it's a lot less of a common pattern for custom volumes than it is for instance. For custom volumes, you expect the frequency to widely vary depending on what the volume is used for, which is less true for instances.

So I think we'll do without it for now and if we see a strong need, can add it.
I didn't really want to have to think about whether to inherit or copy on create as both may be surprising to users :)

Reply:
@stgraber fair enough sounds good.